runtime/interpreter/interpreter.go

```diff
@@ -3059,10 +3059,7 @@ var typeFunction = NewUnmeteredHostFunctionValue(
 		staticType := ConvertSemaToStaticType(invocation.Interpreter, ty)
 		return NewTypeValue(invocation.Interpreter, staticType)
 	},
-	&sema.FunctionType{
```

Comment:
Move to sema, define a function type like for all other run-time type constructors